dowhy/causal_prediction/algorithms/cacm.py

```diff
@@ -0,0 +1,153 @@
+import torch
+from torch.nn import functional as F
+from torch import nn
+from dowhy.causal_prediction.algorithms.base_algorithm import Algorithm
+from dowhy.causal_prediction.algorithms.utils import mmd_compute
+
+class CACM(Algorithm): 
```

Comment:
needs docstring as I described above. 
In addition, here we can add a reference to the arxiv paper in the description of the class/init method